File: Python3/Student Attendance Record I.py
#1
class Solution:
    def checkRecord(self, s: str) -> bool:
        absent = 0
        late = 0

        for c in s:
            if c == 'A':
                absent+=1
                if absent >= 2:
                    return False
            if c == 'L':
                late+=1
                if late >= 3:
                    return False
            else:
                late = 0
        
        return True
            
#2
class Solution:
    def checkRecord(self, s: str) -> bool:
        return (s.count('A') < 2) and ('LLL' not in s)
            

File: Golang/Student Attendance Record I.go
//1
func checkRecord(s string) bool {
    return strings.Count(s, "A") < 2 && strings.Index(s, "LLL") == -1
}

//2
func checkRecord(s string) bool {
    absent := 0

    for _, ch := range s {

        if ch == 'A' {
            absent += 1
        }

        if absent >= 2 {
            return false
        }
    }


    return !strings.Contains(s, "LLL")
}